src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py

```diff
@@ -597,6 +599,84 @@ def _test_private_endpoint_connection(self, resource_group, server, database_eng
                  .format(server_pec_id))
 
 
+class NetworkPrivateLinkBatchAccountScenarioTest(ScenarioTest):
+    def _get_test_data_file(self, filename):
+        filepath = os.path.join(os.path.dirname(os.path.realpath(__file__)), filename)
+        self.assertTrue(os.path.isfile(filepath), 'File {} does not exist.'.format(filepath))
+        return filepath
+
+    # Currently private-link-resource and private-endpoint-connection are whitelist only features so scenario tests are limited
+    @ResourceGroupPreparer(location='westcentralus')
+    def test_private_link_resource_batch_account(self, resource_group, batch_account_name='testprivatelink'):
+        self.kwargs.update({
+            'vnet_name': self.create_random_name('testvnet', 20),
+            'subnet_name': self.create_random_name('testsubnet', 20),
+            'second_endpoint_name': self.create_random_name('priv_endpoint', 25),
+            'second_endpoint_conn_name': self.create_random_name('priv_endpointconn', 25),
+            'approval_desc': 'You are approved!',
+            'rejection_desc': 'You are rejected!',
+            'rg': resource_group,
+            'acc_n': batch_account_name,
+            'loc': 'westcentralus'
+        })
+        account = self.cmd('batch account create -g {rg} -n {acc_n} -l {loc} --public-network-access disabled').assert_with_checks([
+            self.check('name', '{acc_n}'),
+            self.check('location', '{loc}'),
+            self.check('resourceGroup', '{rg}')]).get_output_in_json()
+        self.kwargs['acc_id'] = account['id']
+        # create subnet with disabled endpoint network policies
+        self.cmd('network vnet create -g {rg} -n {vnet_name} --subnet-name {subnet_name}')
+        self.cmd('network vnet subnet update -g {rg} --vnet-name {vnet_name} --name {subnet_name} --disable-private-endpoint-network-policies true')
+        if self.is_live or self.in_recording:
+            import time
+            time.sleep(30)
+
+        # add an endpoint and then reject it
+        self.cmd(
+            'network private-endpoint create '
+            '-n {second_endpoint_name} '
+            '-g {rg} '
+            '--subnet {subnet_name} '
+            '--vnet-name {vnet_name} '
+            '--private-connection-resource-id {acc_id} '
+            '--group-ids batchAccount '
+            '--connection-name {second_endpoint_conn_name} '
+            '--manual-request').get_output_in_json()
+        private_endpoints = self.cmd('network private-endpoint-connection list --name {acc_n} --resource-group {rg} --type Microsoft.Batch/batchAccounts', checks=[
```

Comment:
Does `az batch account show` contain private endpoint or not?

Reply:
Yea just added a basic test